content/en/blog/2023/secure-apps-with-istio/index.md

```diff
@@ -0,0 +1,353 @@
+and negotiate a session key, so that the rest of the connection can be encrypted based on the
+session key. When your applications make a mTLS connection, server and client negotiate a cipher
+suite, which dictates what encryption algorithm your applications will use for the rest of the
+connection and your applications also negotiate the cryptographic session key to use. The whole
+handshake is designed to resist tampering - interference by any entities that do not possess the
+same unique, cryptographically verifiable identity document as the source and/or destination will be
+rejected. For this reason, it is important to check the whole handshake and verify its integrity
+before any communicating peer continues with the application data.
+
+The handshake can be thought of as having three phases per the
+[handshake protocol overview](https://datatracker.ietf.org/doc/html/rfc8446#section-2) in the TLS 1.3
+specification - again, let’s use the example of  a `frontend` application calling a backend
+`checkout` application:
+
+1. Phase 1: `frontend` and `checkout` negotiates the cryptographic parameters and encryption keys
+that can be used to protect the rest of the handshake and traffic data.
+1. Phase 2: everything in this phase and after are encrypted. In this phase, `frontend` and `checkout` establish other handshake parameters, and whether or not the client is also
+authenticated - that is, mTLS.
+1. Phase 3: `frontend` authenticates `checkout` via its cryptographically verifiable identity (and, in mTLS, `checkout` authenticates `frontend` in the same way).
+
+There are a few major differences since TLS 1.2 related to handshake, refer to the TLS 1.3 specification for [more details](https://datatracker.ietf.org/doc/html/rfc8446#section-1.2):
+
+1. All handshake messages (phase 2 and 3) are encrypted **using the encryption keys negotiated in phase 1**.
+1. Legacy symmetric encryption algorithms have been pruned.
+1. A zero round-trip time (0-RTT) mode was added, saving a round trip at connection setup.
+
+#### Record protocol
+
+Having negotiated the TLS protocol version, session-key & [HMAC](https://en.wikipedia.org/wiki/HMAC)
+during the handshake phase, the peers can now securely exchange encrypted data that is chunked by the [record protocol](https://datatracker.ietf.org/doc/html/rfc8446#section-5). It is critical (and
+required as part of the spec) to use the exact same negotiated parameters from the handshake to
+encrypt the traffic to ensure the traffic confidentiality and integrity.
+
+Putting the two protocols from the TLS 1.3 specification together and using the `frontend` and
+`checkout` applications to illustrate the flow as below:
+
+{{< image width="100%"
+    link="mtls-flow.png"
+    caption="mTLS flows when the frontend calls the checkout application"
+    >}}
+
+Who issues the identity certificates for `frontend` and `checkout`? It is commonly issued by a
```

Comment:
```suggestion
Who issues the identity certificates for `frontend` and `checkout`? They are commonly issued by a
```

Reply:
Hmm, are you sure it is "They are" here?  It refers to a CA, i believe.

cc @dhawton who is a native speaker to chime in here.

Reply:
They would be more accurate here due to the plurality of certificates. The subject is the identity certificates for frontend and checkout rather than the CA.